cli/script/command-executor.ts

```diff
@@ -254,8 +269,32 @@ export var deploymentList = (command: cli.IDeploymentListCommand, showPackage: b
                         return deploymentKeys[0].key;
                     });
             });
+            
+            var deploymentKeyList: string[];
             return Q.all(deploymentKeyPromises)
-                .then((deploymentKeyList: string[]) => {
+                .then((retrievedDeploymentKeyList: string[]) => {
+                    deploymentKeyList = retrievedDeploymentKeyList;
+                    if (showPackage) {
+                        var metricsPromises: Promise<void>[] = deployments.map((deployment: Deployment) => {
+                            if (deployment.package) {
+                                return sdk.getDeploymentMetrics(theAppId, deployment.id)
```

Comment:
(Open for discussion) - As with below, would it be worth moving this code onto the server-side to save the round trip?